api/surelia/test/index.js

```diff
@@ -0,0 +1,372 @@
+var hoodiecrow = require("hoodiecrow"),
+    inspect = require('util').inspect;
+var timeout = 500
+
+// Define Hoodiecrow IMAP server
+var server = hoodiecrow({
+  storage: {
+    "INBOX": {
+      messages: [{
+        raw: "Subject: hello 1\r\n\r\nWorld 1!",
+        internaldate: "14-Sep-2013 21:22:28 -0300",
+      }, {
+        raw: "Subject: hello 2\r\n\r\nWorld 2!",
+        flags: ["\\Seen"]
+      }, {
+        raw: "Subject: hello 3\r\n\r\nWorld 3!"
+      }, {
+        raw: "From: sender name <sender@example.com>\r\n" +
+            "To: Receiver name <receiver@example.com>\r\n" +
+            "Subject: hello 4\r\n" +
+            "Message-Id: <abcde>\r\n" +
+            "Date: Fri, 13 Sep 2013 15:01:00 +0300\r\n" +
+            "\r\n" +
+            "World 4!"
+      }, {
+        raw: "Subject: hello 5\r\n\r\nWorld 5!"
+      }, {
+        raw: "Subject: hello 6\r\n\r\nWorld 6!"
+      }]
+    },
+    "": {
+      "separator": "/",
+      "folders": {
+        "[Gmail]": {
+          "flags": ["\\Noselect"],
+          "folders": {
+            "All Mail": {
+                "special-use": "\\All"
+            },
+            "Drafts": {
+                "special-use": "\\Drafts"
+            },
+            "Important": {
+                "special-use": "\\Important"
+            },
+            "Sent Mail": {
+                "special-use": "\\Sent"
+            },
+            "Spam": {
+                "special-use": "\\Junk"
+            },
+            "Starred": {
+                "special-use": "\\Flagged"
+            },
+            "Trash": {
+                "special-use": "\\Trash"
+            }
+          }
+        }
+      }
+    }
+  }
+});
+
+
+var Imap = require(__dirname + "/../../../api/surelia").module.Imap;
+var credentials = {
+  user : "testuser",
+  password : "testpass",
+  host : "localhost",
+  port : 1143,
+  tls : false
+}
+var mail = new Imap(credentials);
+
+var SMTPConnection = require(__dirname + "/../../../api/surelia").module.SMTP;
+var smtp;
+
+// Connect to the server once it is actually listening
+server.listen(1143, function(){
+  
+  // Start unit testing
+  describe("SMTP", function() {
+    describe("SMTP Initial and Auth", function() {
+
+      it("Should connect to SMTP server", function(done){
+        var options = {
+          host : "localhost",
+          port : 2525,
+
+          // Gmail configuration
+          /* host : "smtp.gmail.com", */
+          /* port : 465, */
+          /* requireTLS : true, */
+          /* secure : true */
+        }
+        smtp = new SMTPConnection(options);
+        smtp.connect()
+          .then(function(){
+            done();
+          })
+          .catch(function(err){
+            console.log(err);
+            should(1).equal(2);
+          })
+      })
+      // XXX This unit testing is skipped as the fake smtp server is not supporting auth yet
+      /* it("Should get authenticated to SMTP server", function(done){ */
```

Comment:
Kasih `it.skip('Should .....')` jadi saat UT jalan akan ketahuan bahwa UT ini sengaja dilewati.